contracts/pool/PoolFactory.sol

```diff
@@ -38,7 +38,7 @@ contract PoolFactory is IPoolFactory, Pausable, ReentrancyGuard {
 
     // Instantiates a Pool
     function createPool(
-        string calldata poolId,
+        string memory poolId,
```

Comment:
What have you chosen the `memory` data location for poolId for?
Don't we want to keep `calldata` gas optimization wise?